### PR TITLE
suggest running reentry scan for unstorable Node

### PR DESCRIPTION
**Pull request by ltalirz**
When encountering a Node that isn't storable because of an entry point
that isn't registered, suggest to run `reentry scan`.
We got a user question regarding this, apparently this was not obvious.

### CODE REVIEW

**Review by ramirezfranciscof — APPROVED** [review]
Not sure why you wanted @sphuber in particular to see this, seems pretty straightforward and for me its ready to go (I would maybe say "Consider trying again after running `reentry scan`", but it shouldn't make much of a difference).

I'll leave it to you @ltalirz to merge it or wait for the other review as you prefer.

**Comment by sphuber** [comment]
I didn't add this sentence originally, because it is not guaranteed to be the problem. If the plugin is not even installed, running `reentry scan` will have zero effect. For example, if someone clones the repo of a plugin and then imports one of its data plugins directly through the path and tries to store it, they will be faced with this error and `reentry scan` won't help. So maybe adding 

> Please install the corresponding plugin first and run `reentry scan`

would be more accurate?

**Comment by ramirezfranciscof** [comment]
Mmm a bit verbose but all encompassing option?:

> Please check the installed plugins using `verdi plugin list` and `verdi plugin list <PLUGIN_TYPE>`. If it is not listed there, you may need to install the corresponding plugin and/or run `reentry scan` to update the registry.

File: aiida/orm/nodes/node.py
Comment (ltalirz) [inline aiida/orm/nodes/node.py:201]:
```suggestion
             + ' Consider running `reentry scan`.' \
             + ' If the issue persists, check that the corresponding plugin is installed and that the entry point shows up in `verdi plugin list`'
```

**Comment by ltalirz** [comment]
Thanks to both for your suggestions.

I think in most cases the problem is likely to be the reentry cache (as in the user question I got), so running `reentry scan`  should be the first suggestion.

I've incorporated your suggestions in a follow-up sentence.

**Comment by ramirezfranciscof** [comment]
It works for me, but I was also ok with the previous one so this time I'll let @sphuber have the approving review if he agrees.

P.S.: I think the CI apocalypse is due to an extra space after the line continuation `\` on the third line added (Python is moody, hehe). Is this something we should add to our pre-commit hooks to catch, or do they already have this check?

**Comment by ltalirz** [comment]
> P.S.: I think the CI apocalypse is due to an extra space after the line continuation `\` on the third line added (Python is moody, hehe). Is this something we should add to our pre-commit hooks to catch, or do they already have this check?

Sorry, I thought I might be able to do it from the github interface but it's always easy to make mistakes...

File: aiida/orm/nodes/node.py
Comment (sphuber) [inline aiida/orm/nodes/node.py:204]:
```suggestion
            raise exceptions.StoringNotAllowed(
                f'class `{self.__module__}:{self.__class__.__name__}` does not have a registered entry point. '
                'Consider running `reentry scan`. If the issue persists, check that the corresponding plugin is '
                'installed and that the entry point shows up in `verdi plugin list`.'
            )
```

**Review by sphuber — COMMENTED** [review]
Fine with the new message. Would just change the code to put the message inside the constructor of the exception. That makes all the `\` and `+` characters unnecessary and so there is less clutter.

**Review by sphuber — APPROVED** [review]
Thanks @ltalirz

**Pull request merged**
